src/KatesCausalModel/CausalModels.jl

```diff
@@ -0,0 +1,110 @@
+module CausalModels
+
+using Distributions
+using Statistics
+export ExogenousVariable, EndogenousVariable, intervene, randomsample, prob
+
+# Fill in the rest
+struct ExogenousVariable
+    name::Symbol
+    dist::Bernoulli{Float64}
+end
+
+struct EndogenousVariable
+    logical_operator
+    args::Tuple
+end
+
+struct ModifiedEndoVar
+    endo_var::EndogenousVariable
+    dict::Dict
+end
+
+function (v::EndogenousVariable)(u::NamedTuple{(:courtorder, :nervous),Tuple{Int64,Int64}})
+    total = u.courtorder + u.nervous
+    if total == 0
+        return 0
+    else
+        return 1
+    end
+end
+
+function randomsample(var, calculated=Dict())
+    if typeof(var) == ExogenousVariable 
```

Comment:
You shouldn't need to use `typeof`

`function randomsample(var::ExogenousVariable, ....)`